File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:    
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        nodeArr = []
        for node in lists:
            while node:
                nodeArr.append(node.val)
                node = node.next
        res = ListNode()
        curr = res
        nodeArr = sorted(nodeArr)
        for val in nodeArr:
            curr.next = ListNode(val)
            curr = curr.next
        return res.next
        

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode node = head;
        ArrayList<Integer> visited = new ArrayList<Integer>(10);
        while(node.next != null){
            if(visited.contains(node.val)){
                return true;
            }
            visited.add(node.val);
            node = node.next;
        }
        return false;
    }
}
